source/sessions/snapshot-sessions.rst

```diff
@@ -0,0 +1,339 @@
+
+Applications will start a new client session for snapshot reads like
+this:
+
+.. code:: typescript
+
+    options = new SessionOptions(isSnapshot = true);
+    session = client.startSession(options);
+
+All read operations performed using this session will be read from same snapshot.
+
+If no value is provided for ``isSnapshot`` a value of false is
+implied.
+
+MongoClient changes
+===================
+
+There are no API changes to ``MongoClient`` to support snapshot reads.
+Applications indicate whether they want snapshot reads by setting the
+``isSnapshot`` field in the options passed to the ``startSession`` method.
+
+SessionOptions changes
+======================
+
+``SessionOptions`` change summary
+
+.. code:: typescript
+
+    class SessionOptions {
+        Optional<bool> isSnapshot;
+
+        // other options defined by other specs
+    }
+
+In order to support snapshot reads a new property named
+``isSnapshot`` is added to ``SessionOptions``. Applications set
+``isSnapshot`` when starting a client session to indicate
+whether they want snapshot reads. All read operations performed
+using that client session will share the same snapshot.
+
+Each new member is documented below.
+
+isSnapshot
+---------
+
+Applications set ``isSnapshot`` when starting a session to
+indicate whether they want snapshot reads.
+
+Note that the ``isSnapshot`` property is optional. The default value of
+this property is false.
+
+Snapshot reads and causal consistency are mutually exclusive. Therefore if ``isSnapshot`` is set to true,
+``causalConsistency`` property is set to false. Client MUST throw an Error if both ``isSnapshot`` and ``causalConsistency`` are set to true.
+Snapshot reads are supported both on primaries and secondaries.
+
+MongoDatabase changes
+=====================
+
+There are no additional API changes to ``MongoDatabase`` beyond those specified in
+the Sessions Specification. All ``MongoDatabase`` methods that talk to the server
+have been overloaded to take a session parameter. If that session was started
+with ``isSnapshot = true`` then all read operations using that session will
+will share the same snapshot.
+
+MongoCollection changes
+=======================
+
+There are no additional API changes to ``MongoCollection`` beyond those specified
+in the Sessions Specification. All ``MongoCollection`` methods that talk to the
+server have been overloaded to take a session parameter. If that session was
+started with ``isSnapshot = true`` then all operations using that
+session will share the same snapshot.
+
+ReadConcern changes
+===================
+
+``snapshot`` added to `ReadConcernLevel enumeration <../read-write-concern/read-write-concern.rst#read-concern>`_.`.
+
+Server Commands
+===============
+
+There are no new server commands related to snapshot reads. Instead,
+snapshot reads are implemented by:
+
+1. Saving the ``atClusterTime`` returned by 5.0+ servers for the first find/aggregate operation in a
```

Comment:
What about `distinct`? Or, can only `find` or `aggregate` be the first read operation to fetch and set the `atClusterTime`?